api/src/main/java/org/openmrs/api/impl/AdministrationServiceImpl.java

```diff
@@ -150,7 +152,68 @@ public String getGlobalProperty(String propertyName) throws APIException {
 			return null;
 		}
 		
-		return dao.getGlobalProperty(propertyName);
+		String globalProperty = dao.getGlobalProperty(propertyName);
+		if (StringUtils.isBlank(globalProperty)) {
+			return null;
+		} else {
+			GlobalProperty property = dao.getGlobalPropertyObject(propertyName);
+			if (canViewGlobalProperty(property)) {
+				return globalProperty;
+			} else {
+				throw new APIException("GlobalProperty.error.privilege.required.view", new Object[] {
+					property.getViewPrivilege().getPrivilege(), propertyName });
+			}
+		}
+	}
+	
+	private boolean canViewGlobalProperty(GlobalProperty property) {
+		if (property == null) {
+			throw new IllegalArgumentException("Global property argument can not be null");
+		}
+		User user = Context.getAuthenticatedUser();
+		if (property.getViewPrivilege() == null) {
+			return true;
+		}
+		
+		return user.hasPrivilege(property.getViewPrivilege().getPrivilege());
+	}
+	
+	private boolean canDeleteGlobalProperty(GlobalProperty property) {
+		if (property == null) {
+			throw new IllegalArgumentException("Global property argument can not be null");
+		}
+		User user = Context.getAuthenticatedUser();
+		if (property.getDeletePrivilege() == null) {
```

Comment:
If property.getDeletePrivilege() turns out to be null, why call Context.getAuthenticatedUser()?

Reply:
@dkayiwa thanks for this insight👍🏾 Changed this so that ```Context.getAuthenticatedUser()``` is only called when ```property.getDeletePrivilege()``` is not null.